### DISCUSSION TITLE
Possible with different units for input and output

### DISCUSSION THREAD

**Discussion by Stadly** [langfuse/langfuse#1268] [Ideas]
### Describe the feature or potential improvement

When using models like Whisper, the input and output does not have the same unit. The input may be seconds, while the output is characters. Splitting `unit` into `outputUnit` and `inputUnit` would allow for this.

```js
langfuse.generation({
  usage: {
    input: audio.duration,
    inputUnit: "SECONDS",
    output: transcript.length,
    outputUnit: "CHARACTERS",
  }
});
```

### Additional information

I'm using the JavaScript SDK, version 3.2.0

**Comment by marcklingen** [comment]
Hi @Stadly, thank for the suggestion! Makes sense especially for both TTS and STT. As this needs a partial rewrite of many usage/cost related APIs it is not a small change. Can you solve for this in the meantime by just knowing that the unit deviates in your case?

**Reply by Stadly** [reply]
Yes, it is currently not a blocker for me, but I think it is an important improvement.

**Reply by marcklingen** [reply]
Ok perfect. Thanks for confirming and I totally agree
